src/planner/binder/tableref/bind_basetableref.cpp

```diff
@@ -21,15 +21,19 @@ unique_ptr<BoundTableRef> Binder::Bind(BaseTableRef &ref) {
 	QueryErrorContext error_context(root_statement, ref.query_location);
 	// CTEs and views are also referred to using BaseTableRefs, hence need to distinguish here
 	// check if the table name refers to a CTE
-	auto found_cte = FindCTE(ref.table_name, ref.table_name == alias);
+	auto found_cte = FindCTE(ref.table_name + ref.schema_name, ref.table_name == alias);
```

Comment:
Maybe `FindCTE` should just not be run if the table reference is qualified (i.e. if `ref.schema_name` is not empty)? Adding the schema name to the end might cause weird situations where `schema.table` is equivalent to `tableschema`.